src/compiler/checker.ts

```diff
@@ -29297,6 +29297,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
         function narrowTypeByCallExpression(type: Type, callExpression: CallExpression, assumeTrue: boolean): Type {
             if (hasMatchingArgument(callExpression, reference)) {
+                // Ordinarily we don't need to to this in control flow analysis because the Binder breaks this down.
+                // However, we may encounter the need to narrow this down here when analyzing aliased conditional expressions.

```

Comment:
A need to do some extra narrowings when dealing with aliased conditions isn't new, see: https://github.dev/microsoft/TypeScript/blob/3163fe7e3898c1f48cd9bc097b96e3426cd2a453/src/compiler/checker.ts#L28886-L28888